### ISSUE TITLE
hail ignores the --ignore-soft-errors flag

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1062]
Originally reported of Google Code with ID 1006.

```
hail has a `--ignore-soft-errors` flag that is supposed to override the "N+1 and other soft checks" in at least Ganeti.HTools.Node.addPriEx.

That seems to be ignored in certain cases; functions like `allocateOnSingle` don't have a way to pass it to the eventual Node.hs.

The problematic call order is:

main (in Hail.hs)
runIAllocator
processRequest
tryMGAlloc
findBestAllocGroup
tryAlloc
allocateOnSingle
addPri (does not use `force` parameter)


There is a test for `hail --ignore-soft-errors` in `htools-hail.test` that shows that the flag has an effect sometimes; however, this is because the test takes a different code path that doesn't arrive at `addPri`.


The ghci `:trace` that helped me find this:

[src/Ganeti/HTools/Node.hs:(628,1)-(725,18)] *Main System.Environment> :hist 300
-1  : allocateOnSingle (src/Ganeti/HTools/Cluster.hs:578:14-31)
-2  : >>= (src/Ganeti/BasicTypes.hs:(114,3)-(115,24))
-3  : >>= (src/Ganeti/BasicTypes.hs:115:21-24)
-4  : instMatchesPolicy (src/Ganeti/HTools/Instance.hs:357:10-14)
-5  : diskTemplate (src/Ganeti/HTools/Instance.hs:101:5-16)
-6  : diskTemplate (src/Ganeti/HTools/Instance.hs:101:5-16)
-7  : instMatchesPolicy (src/Ganeti/HTools/Instance.hs:356:6-22)
-8  : iPolicyDiskTemplates (src/Ganeti/HTools/Types.hs:(246,3)-(257,3))
-9  : instMatchesPolicy (src/Ganeti/HTools/Instance.hs:356:31-57)
-10 : instMatchesPolicy (src/Ganeti/HTools/Instance.hs:356:6-57)
-11 : instMatchesPolicy (src/Ganeti/HTools/Instance.hs:(356,3)-(358,23))
-12 : >>= (src/Ganeti/BasicTypes.hs:115:21-24)
-13 : instCompareISpec (src/Ganeti/HTools/Instance.hs:326:17-21)
-14 : iSpecDiskCount (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-15 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:44-65)
-16 : disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-17 : disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-18 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:22-31)
-19 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:14-32)
-20 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:14-65)
-21 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:5-65)
-22 : diskTemplate (src/Ganeti/HTools/Instance.hs:101:5-16)
-23 : diskTemplate (src/Ganeti/HTools/Instance.hs:101:5-16)
-24 : instCompareISpec (src/Ganeti/HTools/Instance.hs:323:5-21)
-25 : instCompareISpec (src/Ganeti/HTools/Instance.hs:323:5-37)
-26 : instCompareISpec (src/Ganeti/HTools/Instance.hs:(323,5)-(324,65))
-27 : iSpecSpindleUse (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-28 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:56-78)
-29 : spindleUse (src/Ganeti/HTools/Instance.hs:102:5-14)
-30 : spindleUse (src/Ganeti/HTools/Instance.hs:102:5-14)
-31 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:30-44)
-32 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:30-78)
-33 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:21-78)
-34 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:5-16)
-35 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:5-78)
-36 : instCompareISpec (src/Ganeti/HTools/Instance.hs:(316,5)-(319,70))
-37 : iSpecCpuCount (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-38 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:35-55)
-39 : vcpus (src/Ganeti/HTools/Instance.hs:93:5-9)
-40 : vcpus (src/Ganeti/HTools/Instance.hs:93:5-9)
-41 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:14-23)
-42 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:14-55)
-43 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:5-55)
-44 : iSpecDiskSize (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-45 : instCompareISpec (src/Ganeti/HTools/Instance.hs:313:34-54)
-46 : dskSize (src/Ganeti/HTools/Instance.hs:82:5-11)
-47 : dskSize (src/Ganeti/HTools/Instance.hs:82:5-11)
-48 : instCompareISpec (src/Ganeti/HTools/Instance.hs:313:23-65)
-49 : disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-50 : disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-51 : instCompareISpec (src/Ganeti/HTools/Instance.hs:314:6-15)
-52 : instCompareISpec (src/Ganeti/HTools/Instance.hs:(313,18)-(314,16))
-53 : instCompareISpec (src/Ganeti/HTools/Instance.hs:(313,5)-(314,16))
-54 : iSpecMemorySize (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-55 : minMaxISpecsMaxSpec (src/Ganeti/HTools/Types.hs:(234,3)-(237,3))
-56 : instMatchesMinMaxSpecs (src/Ganeti/HTools/Instance.hs:340:24-51)
-57 : instCompareISpec (src/Ganeti/HTools/Instance.hs:312:33-55)
-58 : mem (src/Ganeti/HTools/Instance.hs:90:5-7)
-59 : mem (src/Ganeti/HTools/Instance.hs:90:5-7)
-60 : instCompareISpec (src/Ganeti/HTools/Instance.hs:312:14-21)
-61 : instCompareISpec (src/Ganeti/HTools/Instance.hs:312:14-55)
-62 : instCompareISpec (src/Ganeti/HTools/Instance.hs:312:5-55)
-63 : instCompareISpec (src/Ganeti/HTools/Instance.hs:(311,1)-(326,21))
-64 : instMatchesMinMaxSpecs (src/Ganeti/HTools/Instance.hs:340:3-61)
-65 : >>= (src/Ganeti/BasicTypes.hs:115:21-24)
-66 : instCompareISpec (src/Ganeti/HTools/Instance.hs:326:17-21)
-67 : iSpecDiskCount (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-68 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:44-65)
-69 : disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-70 : disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-71 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:22-31)
-72 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:14-32)
-73 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:14-65)
-74 : instCompareISpec (src/Ganeti/HTools/Instance.hs:324:5-65)
-75 : diskTemplate (src/Ganeti/HTools/Instance.hs:101:5-16)
-76 : diskTemplate (src/Ganeti/HTools/Instance.hs:101:5-16)
-77 : instCompareISpec (src/Ganeti/HTools/Instance.hs:323:5-21)
-78 : instCompareISpec (src/Ganeti/HTools/Instance.hs:323:5-37)
-79 : instCompareISpec (src/Ganeti/HTools/Instance.hs:(323,5)-(324,65))
-80 : iSpecSpindleUse (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-81 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:56-78)
-82 : spindleUse (src/Ganeti/HTools/Instance.hs:102:5-14)
-83 : spindleUse (src/Ganeti/HTools/Instance.hs:102:5-14)
-84 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:30-44)
-85 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:30-78)
-86 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:21-78)
-87 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:5-16)
-88 : instCompareISpec (src/Ganeti/HTools/Instance.hs:321:5-78)
-89 : exclStorage (src/Ganeti/HTools/Node.hs:190:5-15)
-90 : exclStorage (src/Ganeti/HTools/Node.hs:190:5-15)
-91 : allocateOnSingle (src/Ganeti/HTools/Cluster.hs:577:55-72)
-92 : instCompareISpec (src/Ganeti/HTools/Instance.hs:(316,5)-(319,70))
-93 : iSpecCpuCount (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-94 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:35-55)
-95 : vcpus (src/Ganeti/HTools/Instance.hs:93:5-9)
-96 : vcpus (src/Ganeti/HTools/Instance.hs:93:5-9)
-97 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:14-23)
-98 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:14-55)
-99 : instCompareISpec (src/Ganeti/HTools/Instance.hs:315:5-55)
-100: iSpecDiskSize (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-101: instCompareISpec (src/Ganeti/HTools/Instance.hs:313:34-54)
-102: dskSize (src/Ganeti/HTools/Instance.hs:82:5-11)
-103: dskSize (src/Ganeti/HTools/Instance.hs:82:5-11)
-104: instCompareISpec (src/Ganeti/HTools/Instance.hs:313:23-65)
-105: parseBaseInstance:disks (src/Ganeti/HTools/Backend/IAlloc.hs:103:15-52)
-106: disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-107: disks (src/Ganeti/HTools/Instance.hs:92:5-9)
-108: instCompareISpec (src/Ganeti/HTools/Instance.hs:314:6-15)
-109: instCompareISpec (src/Ganeti/HTools/Instance.hs:(313,18)-(314,16))
-110: instCompareISpec (src/Ganeti/HTools/Instance.hs:(313,5)-(314,16))
-111: iSpecMemorySize (src/Ganeti/HTools/Types.hs:(182,3)-(195,3))
-112: minMaxISpecsMinSpec (src/Ganeti/HTools/Types.hs:(234,3)-(237,3))
-113: instMatchesMinMaxSpecs (src/Ganeti/HTools/Instance.hs:339:24-51)
-114: instCompareISpec (src/Ganeti/HTools/Instance.hs:312:33-55)
-115: mem (src/Ganeti/HTools/Instance.hs:90:5-7)
-116: mem (src/Ganeti/HTools/Instance.hs:90:5-7)
-117: instCompareISpec (src/Ganeti/HTools/Instance.hs:312:14-21)
-118: instCompareISpec (src/Ganeti/HTools/Instance.hs:312:14-55)
-119: instCompareISpec (src/Ganeti/HTools/Instance.hs:312:5-55)
-120: instCompareISpec (src/Ganeti/HTools/Instance.hs:(311,1)-(326,21))
-121: instMatchesMinMaxSpecs (src/Ganeti/HTools/Instance.hs:339:3-61)
-122: >>= (src/Ganeti/BasicTypes.hs:(114,3)-(115,24))
-123: instMatchesMinMaxSpecs (src/Ganeti/HTools/Instance.hs:(338,47)-(340,61))
-124: instMatchesMinMaxSpecs (src/Ganeti/HTools/Instance.hs:(338,1)-(340,61))
-125: instMatchesSpecs:eithermatch (src/Ganeti/HTools/Instance.hs:349:34-72)
-126: instMatchesSpecs (src/Ganeti/HTools/Instance.hs:348:22-39)
-127: instMatchesSpecs:eithermatch (src/Ganeti/HTools/Instance.hs:(349,9)-(350,35))
-128: instMatchesSpecs (src/Ganeti/HTools/Instance.hs:348:3-49)
-129: iPolicyMinMaxISpecs (src/Ganeti/HTools/Types.hs:(246,3)-(257,3))
-130: iPolicy (src/Ganeti/HTools/Group.hs:59:5-11)
-131: create (src/Ganeti/HTools/Group.hs:(89,3)-(96,9))
-132: create (src/Ganeti/HTools/Group.hs:(88,1)-(96,9))
-133: loadGroup (src/Ganeti/HTools/Backend/Text.hs:214:16-65)
-134: setIdx (src/Ganeti/HTools/Group.hs:102:14-24)
-135: setIdx (src/Ganeti/HTools/Group.hs:102:1-24)
-136: assignIndices:(...) (src/Ganeti/HTools/Loader.hs:138:60-71)
-137: assignIndices (src/Ganeti/HTools/Loader.hs:140:28-57)
-138: group (src/Ganeti/HTools/Node.hs:188:5-9)
-139: group (src/Ganeti/HTools/Node.hs:188:5-9)
-140: setNodePolicy:grp (src/Ganeti/HTools/Loader.hs:171:29-43)
-141: find (src/Ganeti/HTools/Container.hs:79:10-21)
-142: find (src/Ganeti/HTools/Container.hs:79:1-21)
-143: setNodePolicy:grp (src/Ganeti/HTools/Loader.hs:171:13-47)
-144: iPolicy (src/Ganeti/HTools/Group.hs:59:5-11)
-145: setNodePolicy:gpol (src/Ganeti/HTools/Loader.hs:172:14-30)
-146: iPolicy (src/Ganeti/HTools/Node.hs:189:5-11)
-147: idx (src/Ganeti/HTools/Node.hs:154:5-7)
-148: idx (src/Ganeti/HTools/Node.hs:154:5-7)
-149: find (src/Ganeti/HTools/Container.hs:79:10-21)
-150: find (src/Ganeti/HTools/Container.hs:79:1-21)
-151: allocateOnSingle:p (src/Ganeti/HTools/Cluster.hs:574:11-35)
-152: iPolicy (src/Ganeti/HTools/Node.hs:189:5-11)
-153: allocateOnSingle (src/Ganeti/HTools/Cluster.hs:577:38-51)
-154: instMatchesPolicy (src/Ganeti/HTools/Instance.hs:355:26-51)
-155: instMatchesSpecs (src/Ganeti/HTools/Instance.hs:(345,1)-(350,35))
-156: instMatchesPolicy (src/Ganeti/HTools/Instance.hs:355:3-61)
-157: >>= (src/Ganeti/BasicTypes.hs:(114,3)-(115,24))
-158: instMatchesPolicy (src/Ganeti/HTools/Instance.hs:(354,40)-(358,23))
-159: instMatchesPolicy (src/Ganeti/HTools/Instance.hs:(354,1)-(358,23))
-160: allocateOnSingle (src/Ganeti/HTools/Cluster.hs:577:5-73)
-161: >>= (src/Ganeti/BasicTypes.hs:(114,3)-(115,24))
-162: allocateOnSingle (src/Ganeti/HTools/Cluster.hs:(576,6)-(581,49))
-163: allocateOnSingle (src/Ganeti/HTools/Cluster.hs:(573,1)-(581,49))
-164: tryAlloc:sols (src/Ganeti/HTools/Cluster.hs:879:46-69)
-165: concatAllocs (src/Ganeti/HTools/Cluster.hs:(779,1)-(792,70))
-166: tryAlloc:sols (src/Ganeti/HTools/Cluster.hs:879:24-42)
-167: tryAlloc:sols (src/Ganeti/HTools/Cluster.hs:879:24-69)
-168: tryAlloc:sols (src/Ganeti/HTools/Cluster.hs:(878,14)-(880,50))
-169: annotateSolution (src/Ganeti/HTools/Cluster.hs:825:23-67)
-170: annotateSolution (src/Ganeti/HTools/Cluster.hs:825:1-67)
-171: tryAlloc (src/Ganeti/HTools/Cluster.hs:881:15-35)
-172: asSolution (src/Ganeti/HTools/Cluster.hs:127:5-14)
-173: filterMGResults:fn (src/Ganeti/HTools/Cluster.hs:903:33-46)
-174: filterMGResults:fn (src/Ganeti/HTools/Cluster.hs:903:22-47)
-175: return (src/Ganeti/BasicTypes.hs:116:12-13)
-176: tryAlloc (src/Ganeti/HTools/Cluster.hs:881:6-35)
-177: loadNode (src/Ganeti/HTools/Backend/Text.hs:250:59-67)
-178: offline (src/Ganeti/HTools/Node.hs:179:5-11)
-179: offline (src/Ganeti/HTools/Node.hs:179:5-11)
-180: idx (src/Ganeti/HTools/Node.hs:154:5-7)
-181: setXmem (src/Ganeti/HTools/Node.hs:399:17-32)
-182: setXmem (src/Ganeti/HTools/Node.hs:399:1-32)
-183: checkData:newn (src/Ganeti/HTools/Loader.hs:336:39-65)
-184: setFmem (src/Ganeti/HTools/Node.hs:553:6-57)
-185: setFmem (src/Ganeti/HTools/Node.hs:(550,1)-(553,57))
-186: checkData:newn (src/Ganeti/HTools/Loader.hs:(336,25)-(337,50))
-187: idx (src/Ganeti/HTools/Node.hs:154:5-7)
-188: splitCluster:nidxs (src/Ganeti/HTools/Cluster.hs:1754:24-41)
-189: splitCluster:nodes' (src/Ganeti/HTools/Cluster.hs:1755:25-39)
-190: splitCluster (src/Ganeti/HTools/Cluster.hs:1757:22-46)
-191: genAllocNodes:all_nodes (src/Ganeti/HTools/Cluster.hs:847:31-42)
-192: genAllocNodes:filter_fn (src/Ganeti/HTools/Cluster.hs:(843,19)-(846,27))
-193: genAllocNodes:all_nodes (src/Ganeti/HTools/Cluster.hs:847:19-42)
-194: genAllocNodes (src/Ganeti/HTools/Cluster.hs:854:23-44)
-195: genAllocNodes (src/Ganeti/HTools/Cluster.hs:854:17-45)
-196: tryAlloc (src/Ganeti/HTools/Cluster.hs:(865,1)-(881,35))
-197: findBestAllocGroup:sols (src/Ganeti/HTools/Cluster.hs:977:25-43)
-198: >>= (src/Ganeti/BasicTypes.hs:115:21-24)
-199: genAllocNodes (src/Ganeti/HTools/Cluster.hs:854:13-46)
-200: genAllocNodes (src/Ganeti/HTools/Cluster.hs:(853,6)-(856,73))
-201: genAllocNodes (src/Ganeti/HTools/Cluster.hs:(842,1)-(856,73))
-202: findBestAllocGroup:sols (src/Ganeti/HTools/Cluster.hs:976:25-55)
-203: >>= (src/Ganeti/BasicTypes.hs:(114,3)-(115,24))
-204: findBestAllocGroup:sols (src/Ganeti/HTools/Cluster.hs:(976,25)-(977,43))
-205: filterMGResults:fn (src/Ganeti/HTools/Cluster.hs:(901,11)-(905,49))
-206: findBestAllocGroup:sols (src/Ganeti/HTools/Cluster.hs:(976,20)-(977,44))
-207: findBestAllocGroup:groups (src/Ganeti/HTools/Cluster.hs:970:34-61)
-208: splitCluster (src/Ganeti/HTools/Cluster.hs:1757:15-59)
-209: computeGroups (src/Ganeti/HTools/Node.hs:1037:18-38)
-210: filterMGResults:fn (src/Ganeti/HTools/Cluster.hs:(900,9)-(905,49))
-211: filterValidGroups (src/Ganeti/HTools/Cluster.hs:930:13-24)
-212: network (src/Ganeti/HTools/Nic.hs:72:5-11)
-213: create (src/Ganeti/HTools/Nic.hs:(91,3)-(97,7))
-214: create (src/Ganeti/HTools/Nic.hs:(90,1)-(97,7))
-215: parseNic (src/Ganeti/HTools/Backend/IAlloc.hs:85:11-52)
-216: network (src/Ganeti/HTools/Nic.hs:72:5-11)
-217: hasRequiredNetworks:hasNetwork (src/Ganeti/HTools/Cluster.hs:920:22-58)
-218: hasRequiredNetworks:hasNetwork (src/Ganeti/HTools/Cluster.hs:920:22-72)
-219: nics (src/Ganeti/HTools/Instance.hs:106:5-8)
-220: create (src/Ganeti/HTools/Instance.hs:(192,3)-(212,12))
-221: create (src/Ganeti/HTools/Instance.hs:(190,1)-(212,12))
-222: parseBaseInstance (src/Ganeti/HTools/Backend/IAlloc.hs:(114,6)-(115,32))
-223: parseData:io (src/Ganeti/HTools/Backend/IAlloc.hs:229:63-70)
-224: updateExclTags (src/Ganeti/HTools/Loader.hs:180:6-42)
-225: updateExclTags (src/Ganeti/HTools/Loader.hs:(177,1)-(180,42))
-226: parseData:io (src/Ganeti/HTools/Backend/IAlloc.hs:229:24-59)
-227: parseData:io (src/Ganeti/HTools/Backend/IAlloc.hs:229:24-70)
-228: nics (src/Ganeti/HTools/Instance.hs:106:5-8)
-229: hasRequiredNetworks (src/Ganeti/HTools/Cluster.hs:919:26-39)
-230: hasRequiredNetworks (src/Ganeti/HTools/Cluster.hs:919:26-55)
-231: hasRequiredNetworks (src/Ganeti/HTools/Cluster.hs:(919,1)-(920,72))
-232: filterValidGroups (src/Ganeti/HTools/Cluster.hs:929:9-41)
-233: filterValidGroups (src/Ganeti/HTools/Cluster.hs:(929,6)-(934,38))
-234: splitCluster:ngroups (src/Ganeti/HTools/Cluster.hs:1752:37-54)
-235: computeGroups:nodes' (src/Ganeti/HTools/Node.hs:1033:16-45)
-236: computeGroups:nodes'' (src/Ganeti/HTools/Node.hs:1034:17-48)
-237: computeGroups (src/Ganeti/HTools/Node.hs:1037:6-47)
-238: computeGroups (src/Ganeti/HTools/Node.hs:(1032,1)-(1037,47))
-239: splitCluster:ngroups (src/Ganeti/HTools/Cluster.hs:1752:17-55)
-240: splitCluster (src/Ganeti/HTools/Cluster.hs:(1753,6)-(1757,68))
-241: splitCluster (src/Ganeti/HTools/Cluster.hs:(1751,1)-(1757,68))
-242: findBestAllocGroup:groups_by_idx (src/Ganeti/HTools/Cluster.hs:969:23-44)
-243: findBestAllocGroup:groups (src/Ganeti/HTools/Cluster.hs:970:16-76)
-244: findBestAllocGroup:groups' (src/Ganeti/HTools/Cluster.hs:(971,17)-(973,28))
-245: filterValidGroups (src/Ganeti/HTools/Cluster.hs:(926,1)-(934,38))
-246: findBestAllocGroup:(...) (src/Ganeti/HTools/Cluster.hs:974:39-68)
-247: findBestAllocGroup:sols (src/Ganeti/HTools/Cluster.hs:(975,14)-(978,60))
-248: findBestAllocGroup:goodSols (src/Ganeti/HTools/Cluster.hs:980:18-37)
-249: sortMGResults (src/Ganeti/HTools/Cluster.hs:914:6-37)
-250: sortMGResults (src/Ganeti/HTools/Cluster.hs:(910,1)-(914,37))
-251: findBestAllocGroup:sortedSols (src/Ganeti/HTools/Cluster.hs:981:20-41)
-252: findBestAllocGroup (src/Ganeti/HTools/Cluster.hs:(982,6)-(988,78))
-253: findBestAllocGroup (src/Ganeti/HTools/Cluster.hs:(968,1)-(988,78))
-254: tryMGAlloc (src/Ganeti/HTools/Cluster.hs:999:7-56)
-255: >>= (src/Ganeti/BasicTypes.hs:(114,3)-(115,24))
-256: tryMGAlloc (src/Ganeti/HTools/Cluster.hs:(997,38)-(1002,47))
-257: tryMGAlloc (src/Ganeti/HTools/Cluster.hs:(997,1)-(1002,47))
-258: processRequest (src/Ganeti/HTools/Backend/IAlloc.hs:419:10-44)
-259: >>= (src/Ganeti/BasicTypes.hs:(114,3)-(115,24))
-260: processRequest (src/Ganeti/HTools/Backend/IAlloc.hs:419:10-66)
-261: parseData (src/Ganeti/HTools/Backend/IAlloc.hs:230:37-70)
-262: parseData (src/Ganeti/HTools/Backend/IAlloc.hs:230:24-71)
-263: parseData (src/Ganeti/HTools/Backend/IAlloc.hs:271:17-36)
-264: wrapReadRequest:(...) (src/Ganeti/HTools/Program/Hail.hs:84:27-28)
-265: loadExternalData (src/Ganeti/HTools/ExtLoader.hs:150:10-30)
-266: wrapReadRequest (src/Ganeti/HTools/Program/Hail.hs:85:16-32)
-267: processRequest:(...) (src/Ganeti/HTools/Backend/IAlloc.hs:416:51-57)
-268: processRequest (src/Ganeti/HTools/Backend/IAlloc.hs:(417,6)-(431,64))
-269: processRequest (src/Ganeti/HTools/Backend/IAlloc.hs:(415,1)-(431,64))
-270: runIAllocator:(...) (src/Ganeti/HTools/Backend/IAlloc.hs:449:14-40)
-271: runIAllocator:(...) (src/Ganeti/HTools/Backend/IAlloc.hs:(449,9)-(452,76))
-272: formatResponse:e_success (src/Ganeti/HTools/Backend/IAlloc.hs:279:31-46)
-273: formatResponse:e_success (src/Ganeti/HTools/Backend/IAlloc.hs:279:19-47)
-274: formatResponse (src/Ganeti/HTools/Backend/IAlloc.hs:282:21-57)
-275: formatResponse (src/Ganeti/HTools/Backend/IAlloc.hs:282:6-57)
-276: formatResponse (src/Ganeti/HTools/Backend/IAlloc.hs:(278,1)-(282,57))
-277: runIAllocator:rstring (src/Ganeti/HTools/Backend/IAlloc.hs:453:17-45)
-278: runIAllocator (src/Ganeti/HTools/Backend/IAlloc.hs:454:6-21)
-279: runIAllocator (src/Ganeti/HTools/Backend/IAlloc.hs:(447,1)-(454,21))
-280: main:(...) (src/Ganeti/HTools/Program/Hail.hs:114:26-72)
-281: main (src/Ganeti/HTools/Program/Hail.hs:116:3-15)
-282: maybeSaveData (src/Ganeti/HTools/ExtLoader.hs:158:31-39)
-283: optSaveCluster (src/Ganeti/HTools/CLI.hs:181:5-18)
-284: optSaveCluster (src/Ganeti/HTools/CLI.hs:181:5-18)
-285: main:savecluster (src/Ganeti/HTools/Program/Hail.hs:97:21-39)
-286: maybeSaveData (src/Ganeti/HTools/ExtLoader.hs:(158,1)-(164,22))
-287: main (src/Ganeti/HTools/Program/Hail.hs:112:3-70)
-288: maybePrintNodes (src/Ganeti/HTools/CLI.hs:797:31-39)
-289: optShowNodes (src/Ganeti/HTools/CLI.hs:186:5-16)
-290: optShowNodes (src/Ganeti/HTools/CLI.hs:186:5-16)
-291: main:shownodes (src/Ganeti/HTools/Program/Hail.hs:95:19-35)
-292: maybePrintNodes (src/Ganeti/HTools/CLI.hs:(797,1)-(801,30))
-293: main (src/Ganeti/HTools/Program/Hail.hs:(109,3)-(110,43))
-294: main (src/Ganeti/HTools/Program/Hail.hs:106:9-19)
-295: main (src/Ganeti/HTools/Program/Hail.hs:106:3-20)
-296: main (src/Ganeti/HTools/Program/Hail.hs:(106,3)-(107,23))
-297: main (src/Ganeti/HTools/Program/Hail.hs:(106,3)-(107,65))
-298: optVerbose (src/Ganeti/HTools/CLI.hs:194:5-14)
-299: optVerbose (src/Ganeti/HTools/CLI.hs:194:5-14)
-300: main:verbose (src/Ganeti/HTools/Program/Hail.hs:96:17-31)
```
Originally added on 2014-12-08 17:14:13 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-12-08 17:27:10 +0000 UTC.

Changed State: Started

**Comment by Ganeti-Issues-Migrator** [comment]
```
--ignore-soft-errors was introduced in 2.13
```
Originally added on 2014-12-08 18:22:51 +0000 UTC.

Added to Milestone: Release2.13
Removed from Milestone: Release2.12

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-12-08 18:23:17 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Taking over, as this essentially blocks issue 999
```
Originally added on 2015-01-13 14:47:54 +0000 UTC.

Mentioned Issue with GoogleCode ID 999 has been migrated to GitHub with ID #1055.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Related, from the patch series fixing issue 999.

commit da42d73baf8b4565b230277780012bf43774a12b
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Jan 13 16:42:39 2015 +0100

    Pass on options in nodeEvacInstance for ChangePrimary
    
    In this way, if --ignore-soft-errors is given to hail, that
    option is also honored when moving out the primary of a DRBD
    instance.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2015-01-14 17:59:36 +0000 UTC.

Mentioned Issue with GoogleCode ID 999 has been migrated to GitHub with ID #1055.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Already in:

commit 6aa2f1f5d3a7036c9fa4772205c9c50207405add
Author: Niklas Hambuechen <niklash@google.com>
Date:   Fri Dec 12 17:00:25 2014 +0100

    HTools: Pass --ignore-soft-errors all the way through
    
    Fixes hail silently ignoring the --ignore-soft-errors flag.
    
    See issue 1006.
    
    For hspace and tests, the default is to use defaultOptions
    (do not force, like it was before).
    
    The core change is in `allocateOnSingle`/`allocateOnPair`,
    which now call `addPriEx force` instead of `addPri` (that one
    defaults to `force = False`).
    
    Signed-off-by: Niklas Hambuechen <niklash@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

```
Originally added on 2015-01-15 16:16:00 +0000 UTC.

Mentioned Issue with GoogleCode ID 1006 has been migrated to GitHub with ID #1062.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit fbe43b73887a151ccc2c42f5a88127057fb357a8
Author: Klaus Aehlig <aehlig@google.com>
Date:   Mon Jan 19 10:52:56 2015 +0100

    Add a test verifying --ignore-soft-errors in hail
    
    In this example, the instance cannot be placed without
    violating exclusion tags. As those are, however, soft
    errors, adding --ignore-soft-errors, the placement is
    possible.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 12d2cbd55f6b037102f192692d93462aa7f20db4
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Jan 13 16:42:39 2015 +0100

    Pass on options in nodeEvacInstance for ChangePrimary
    
    In this way, if --ignore-soft-errors is given to hail, that
    option is also honored when moving out the primary of a DRBD
    instance.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>
    
    Cherry-picked-from: da42d73baf8b4565b230277780012bf43774a12b
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 3c3f9cf634672561fbc1fc2303548f2841c92062
Author: Niklas Hambuechen <niklash@google.com>
Date:   Fri Dec 12 17:00:25 2014 +0100

    HTools: Pass --ignore-soft-errors all the way through
    
    Fixes hail silently ignoring the --ignore-soft-errors flag.
    
    See issue 1006.
    
    For hspace and tests, the default is to use defaultOptions
    (do not force, like it was before).
    
    The core change is in `allocateOnSingle`/`allocateOnPair`,
    which now call `addPriEx force` instead of `addPri` (that one
    defaults to `force = False`).
    
    Signed-off-by: Niklas Hambuechen <niklash@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>
    
    Cherry-picked-from: 6aa2f1f5d3a7036c9fa4772205c9c50207405add
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit d325dd79481aa5714427418f1bbec1fb675161b5
Author: Klaus Aehlig <aehlig@google.com>
Date:   Mon Jan 12 17:34:24 2015 +0100

    Make tags a soft error
    
    Consider exclusion tags violations a soft error. While it
    is not desirable to have to instances providing the same
    service running on the same node, this is still better
    than running on an offline node.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Aaron Karper <akarper@google.com>
    
    Cherry-picked-from: b3ef1e57dbc4d193e475fb03fee307e02d32d02b
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2015-01-19 17:22:40 +0000 UTC.

Mentioned Issue with GoogleCode ID 1006 has been migrated to GitHub with ID #1062.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:03 +0000 UTC.

Changed State: Released
